### PR TITLE
rootless: fix usage on Fedora Silverblue/CoreOS

### PR DESCRIPTION
**Pull request by giuseppe**
here the home directory is a symlink to `/var/home` and runc does like symlinks into the rootfs path.

Resolve the path to the storage before using it.

As part of the PR, also fix the code propagation from the re-exec'ed podman.

### CODE REVIEW

File: pkg/rootless/rootless_linux.go [hunk 1]
```diff
@@ -68,35 +68,35 @@ func tryMappingTool(tool string, pid int, hostID int, mappings []idtools.IDMap)
 }
 
 // BecomeRootInUserNS re-exec podman in a new userNS
```
Comment (mheon) [inline pkg/rootless/rootless_linux.go]: Can you expand the comment to explain what's being returned? Not entirely obvious at first glance that the int is a return code

File: pkg/rootless/rootless_unsupported.go [hunk 2]
```diff
@@ -13,8 +13,8 @@ func IsRootless() bool {
 
 // BecomeRootInUserNS is a stub function that always returns false and an
 // error on unsupported OS's
-func BecomeRootInUserNS() (bool, error) {
-	return false, errors.New("this function is not supported on this os")
+func BecomeRootInUserNS() (bool, ret, error) {
```
[the remark this answers is not in the document]
Comment (giuseppe) [reply pkg/rootless/rootless_unsupported.go]: yes of course, I've updated the PR

**Comment by giuseppe** [comment]
I've missed the notifications on this one, just updated the PR with the last comments I've got.

**Comment by rhatdan** [comment]
LGTM
@mheon @baude @giuseppe @umohnani8 PTAL

**Comment by mheon** [comment]
LGTM
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
Homu seems to be hung, forcing merge.

**Pull request merged**